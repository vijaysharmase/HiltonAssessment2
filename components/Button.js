import React from "react";
import styled from "styled-components";

const StyledButton = styled.input.attrs({ type: "button" })`
	font: 0.8em "Montserrat", sans-serif;
	border-radius: 3px;
	height: 2em;
	width: 5em;
	background-color: #e7e7e7;
`;

const Button = ({ ...props }) => <StyledButton {...props} />;

export default Button;
