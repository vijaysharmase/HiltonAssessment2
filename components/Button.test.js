import React from "react";
import enzyme from "enzyme";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import Button from "./Button";
enzyme.configure({ adapter: new Adapter() });

describe("Button component", () => {
	var clicked = false;
	const props = {
		value: "Submit",
		onClick: () => {
			clicked = true;
		}
	};

	it("check button click functionality", () => {
		const wrapper = enzyme.mount(<Button {...props} />);
		wrapper.find('input[type="button"]').simulate("click");
		expect(clicked).toBe(true);
	});

	it("Compare snapshots", () => {
		const wrapper = renderer.create(<Button {...props} />);
		const button = wrapper.toJSON();
		expect(button).toMatchSnapshot();
	});
});
