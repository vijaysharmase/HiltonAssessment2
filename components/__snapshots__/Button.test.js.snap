// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Compare snapshots 1`] = `
<input
  className="sc-bdVaJa eCboiN"
  onClick={[Function]}
  type="button"
  value="Submit"
/>
`;
